state-chain/pallets/cf-ingress-egress/src/boost_pool/tests.rs

```diff
@@ -160,19 +162,82 @@ fn boosting_with_fees() {
 
 	check_pool(&pool, [(BOOSTER_1, 1000), (BOOSTER_2, 2000)]);
 
-	assert_eq!(pool.provide_funds_for_boosting(BOOST_1, 1010), Ok((1010, 10)));
+	assert_eq!(pool.provide_funds_for_boosting(BOOST_1, 1010, NO_DEDUCTION), Ok((1010, 10)));
 
 	// The recorded amounts include fees (1 is missing due to rounding errors in *test* code)
 	check_pending_boosts(
 		&pool,
 		[(BOOST_1, vec![(BOOSTER_1, 333 + 3, 3), (BOOSTER_2, 667 + 6, 6)])],
 	);
 
-	assert_eq!(pool.process_deposit_as_finalised(BOOST_1), vec![]);
+	assert_eq!(
+		pool.process_deposit_as_finalised(BOOST_1),
+		DepositFinalisationOutcomeForPool { amount_credited: 1010, unlocked_funds: vec![] }
+	);
 
 	check_pool(&pool, [(BOOSTER_1, 1003), (BOOSTER_2, 2006)]);
 }
 
+#[test]
+fn boosting_with_fees_including_network_fee_portion() {
+	const NETWORK_FEE_PORTION_PERCENT: u8 = 30;
+
+	let mut pool = TestPool::new(100); // 1%
+
+	pool.add_funds(BOOSTER_1, 1000);
+	pool.add_funds(BOOSTER_2, 2000);
+
+	check_pool(&pool, [(BOOSTER_1, 1000), (BOOSTER_2, 2000)]);
+
+	const PROVIDED_AMOUNT: u128 = 1000;
+	const FULL_BOOST_FEE: u128 = 10;
+
+	const DEPOSIT_AMOUNT: u128 = PROVIDED_AMOUNT + FULL_BOOST_FEE;
+
+	// NOTE: Full 1% boost fee is charged from the deposit
+	assert_eq!(
+		pool.provide_funds_for_boosting(
+			BOOST_1,
+			DEPOSIT_AMOUNT,
+			Percent::from_percent(NETWORK_FEE_PORTION_PERCENT)
+		),
+		Ok((DEPOSIT_AMOUNT, FULL_BOOST_FEE))
+	);
+
+	const BOOSTER_1_FEE: u128 = 2;
+	const BOOSTER_2_FEE: u128 = 4;
+
+	const TOTAL_BOOSTERS_FEE: u128 = BOOSTER_1_FEE + BOOSTER_2_FEE;
+
+	const NETWORK_FEE_FROM_BOOST: u128 = FULL_BOOST_FEE * NETWORK_FEE_PORTION_PERCENT as u128 / 100;
+
+	// NOTE: we subtract 1 to account for a rounding "error" (in real code any remaining fee will be
+	// used as network fee, so all atomic units will be accounted for)
+	assert_eq!(TOTAL_BOOSTERS_FEE, FULL_BOOST_FEE - NETWORK_FEE_FROM_BOOST - 1);
```

Comment:
Imo, this is a little strange that we're testing something that's explicitly stated can't happen in the real code. Seems like we could write this test in such a way that demonstrates that the statement in the comment is true, maybe capturing init_swap_request's input amount?